dcos_test_utils/gce.py

```diff
@@ -0,0 +1,254 @@
+INSTANCE_TEMPLATE = """
+- type: compute.v1.instanceTemplate
+  name: {name}
+  metadata:
+    dependsOn:
+    - {network}
+  properties:
+    project: {project}
+    properties:
+      machineType: {machineType}
+      disks:
+      - deviceName: boot
+        type: PERSISTENT
+        boot: true
+        autoDelete: true
+        initializeParams:
+          sourceImage: projects/{imageProject}/global/images/family/{sourceImage}
+      networkInterfaces:
+      - network: global/networks/{network}
+        # Access Config required to give the instance a public IP address
+        accessConfigs:
+        - name: External NAT
+          type: ONE_TO_ONE_NAT
+      metadata:
+        items:
+        - key: ssh-keys
+          value: {ssh_user}:{ssh_public_key}"""
+
+# template for a network resource in a gce deployment
+NETWORK_TEMPLATE = """
+- type: compute.v1.network
+  name: {name}
+  properties:
+    autoCreateSubnetworks: True"""
+
+# template for an instance group manager resource in a gce deployment
+MANAGED_INSTANCE_GROUP_TEMPLATE = """
+- type: compute.v1.instanceGroupManager
+  name: {name}
+  metadata:
+    dependsOn:
+    - {instance_template_name}
+  properties:
+    baseInstanceName: vm
+    instanceTemplate: global/instanceTemplates/{instance_template_name}
+    zone: {zone}
+    targetSize: {size}"""
+
+# template for a firewall in the network of a gce deployment
+FIREWALL_TEMPLATE = """
+- type: compute.v1.firewall
+  name: {name}
+  metadata:
+    dependsOn:
+    - {network}
+  properties:
+    description: allow all ports
+    network: global/networks/{network}
+    sourceRanges:
+    - 0.0.0.0/0
+    allowed:
+    - IPProtocol: tcp
+    - IPProtocol: udp
+    - IPProtocol: icmp
+    - IPProtocol: sctp"""
+
+
+# Function decorator that adds detail to potential googleapiclient.errors.HttpError exceptions with code 404 or 409
+def catch_http_exceptions(f):
+    @wraps(f)
+    def handle_exception(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except HttpError as e:
+            if e.resp.status == 404:
+                log.exception("The resource you are trying to access doesn't exist")
+            elif e.resp.status == 409:
+                log.exception('''The specified resources exist and might be under an active operation
+                                   (operation conflict)''')
+            raise e
+
+    return handle_exception
+
+
+class GceWrapper:
+    @catch_http_exceptions
+    def __init__(self, credentials_dict):
+        credentials = GoogleCredentials.get_application_default()
+        self.compute = discovery.build('compute', 'v1', credentials=credentials)
+        self.deployment_manager = discovery.build('deploymentmanager', 'v2', credentials=credentials)
+        self.project_id = credentials_dict['project_id']
+
+    @catch_http_exceptions
+    def create_deployment(self, deployment_name, resources: str):
```

Comment:
This can just be included with the the create method (which I commented should be made into a class method) within the Deployment object. No reason I see to separate this into the GceWrapper class